src/TestableHttpClient/Utils/CancellationTokenExtensions.cs

```diff
@@ -0,0 +1,29 @@
+using System.Reflection;
+
+namespace TestableHttpClient.Utils;
+
+internal static class CancellationTokenExtensions
+{
+    private static FieldInfo? cancellationSourceField;
+
+    private static FieldInfo? CancellationSourceField
+    {
+        get
+        {
+            if (cancellationSourceField is null)
+            {
+                cancellationSourceField = typeof(CancellationToken).GetFields(BindingFlags.Instance | BindingFlags.NonPublic | BindingFlags.Public | BindingFlags.DeclaredOnly).FirstOrDefault(x => x.FieldType == typeof(CancellationTokenSource));
```

Comment:
Consider using `Single` instead, since we can fail hard when we can't find a field.